### PR TITLE
ToDoApp main functionality

### PR DESCRIPTION
**Pull request by o-ivankova**

### CODE REVIEW

**Comment by o-ivankova** [comment]
outdated

File: src/redux/store.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { configureStore } from "@reduxjs/toolkit";
+import {persistedReducer} from "./toDoListSlice";
+import { persistStore } from 'redux-persist'
```
Comment (aabrazouski) [inline src/redux/store.js:3]: You should implement your own logic to persist redux data without using external libraries

File: src/redux/toDoListSlice.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+export const toDoListSlice = createSlice({
+  name: "toDoList",
+  initialState,
+  reducers: {
+    addElement: (state, action) => {
+      const input = action.payload;
+
+      if (input !== "") {
+        state.elementId += 1;
```
Comment (aabrazouski) [inline src/redux/toDoListSlice.js]: You can use package https://www.npmjs.com/package/uuid to generate id.

File: src/components/ToDoListItem/ToDoListItem.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import "./ToDoListItem.scss";
+import { BsX } from "react-icons/bs";
+import PropTypes from "prop-types";
+
+const ToDoListItem = ({ content, completed, onDeleted, onCheckCompleted }) => {
+  return (
+    <div className={`todo-list-item ${completed && " completed"} d-flex`}>
+      <div className="todo-list-item-content" onClick={onCheckCompleted}>
+        <input type="checkbox" checked={completed} />
```
Comment (aabrazouski) [inline src/components/ToDoListItem/ToDoListItem.js]: Create `Checkbox` component please.

File: src/components/ToDoListItem/ToDoListItem.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import "./ToDoListItem.scss";
+import { BsX } from "react-icons/bs";
+import PropTypes from "prop-types";
+
+const ToDoListItem = ({ content, completed, onDeleted, onCheckCompleted }) => {
+  return (
+    <div className={`todo-list-item ${completed && " completed"} d-flex`}>
+      <div className="todo-list-item-content" onClick={onCheckCompleted}>
```
Comment (aabrazouski) [inline src/components/ToDoListItem/ToDoListItem.js]: div is not an interactive element and you should avoid any event handlers on it. It break accessibility of your application
Handle it via `onChange` checkbox event

File: src/components/ToDoListItem/ToDoListItem.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import "./ToDoListItem.scss";
+import { BsX } from "react-icons/bs";
+import PropTypes from "prop-types";
+
+const ToDoListItem = ({ content, completed, onDeleted, onCheckCompleted }) => {
+  return (
+    <div className={`todo-list-item ${completed && " completed"} d-flex`}>
```
Comment (aabrazouski) [inline src/components/ToDoListItem/ToDoListItem.js]: Use package https://www.npmjs.com/package/classnames to handle classnames correctly.

File: src/components/ToDoList/ToDoList.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+  };
+
+  const filteredElements = filterElements(elements, filter);
+
+  const list = filteredElements.map((el) => {
+    const { id } = el;
+
+    const index = elements.findIndex((el) => el.id === id);
+
+    const onDeleted = (id) => {
```
Comment (aabrazouski) [inline src/components/ToDoList/ToDoList.js]: There is no need to create function on each iteration. Just move it outside `map`

File: src/components/ToDoList/ToDoList.js [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  reorderElements,
+} from "../../redux/toDoListSlice";
+import { DragDropContext, Droppable, Draggable } from "react-beautiful-dnd";
+
+const ToDoList = () => {
+  const elements = useSelector((state) => state.toDoList.elements);
+  const filter = useSelector((state) => state.toDoList.filter);
+  const dispatch = useDispatch();
+
+  const filterElements = (elements, filter) => {
```
Comment (aabrazouski) [inline src/components/ToDoList/ToDoList.js:17]: You can move it to selector using https://redux-toolkit.js.org/api/createSelector

File: src/components/AddItemPanel/AddItemPanel.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  };
+
+  const addItem = (input) => {
+    dispatch(addElement(input));
+  };
+
+  return (
+    <div className='add-item-panel'>
+    <form className="add-item-form" onSubmit={onSubmit}>
+      <input
```
Comment (aabrazouski) [inline src/components/AddItemPanel/AddItemPanel.js:28]: All elements like input, button better to move to separate components

File: src/components/ToDoListItem/ToDoListItem.js (hunk 4 above)
Comment (o-ivankova) [reply src/components/ToDoListItem/ToDoListItem.js]: resolved

File: src/components/ToDoListItem/ToDoListItem.js (hunk 3 above)
Comment (o-ivankova) [reply src/components/ToDoListItem/ToDoListItem.js]: resolved

File: src/redux/toDoListSlice.js (hunk 2 above)
Comment (o-ivankova) [reply src/redux/toDoListSlice.js]: resolved

File: src/components/ToDoList/ToDoList.js (hunk 6 above)
Comment (o-ivankova) [reply src/components/ToDoList/ToDoList.js]: resolved

File: src/redux/store.js (hunk 1 above)
Comment (o-ivankova) [reply src/redux/store.js:3]: resolved

File: src/components/ToDoListItem/ToDoListItem.js (hunk 5 above)
Comment (o-ivankova) [reply src/components/ToDoListItem/ToDoListItem.js]: resolved

File: src/components/ToDoList/ToDoList.js (hunk 7 above)
Comment (o-ivankova) [reply src/components/ToDoList/ToDoList.js:17]: resolved

**Pull request merged**